frontend/src/store/slices/authSlice.ts

```diff
@@ -0,0 +1,27 @@
+/* eslint-disable no-console */
+import {  StateCreator } from 'zustand'
+import {instance} from '../../axios'
+import { getHeaders } from '@/utils/getHeaders'
+
+
+const apiRoute = `/auth`
+
+export interface AuthSlice {
+  data: object | null
+  registerUser: (payload:object | null) => void
+  login: (payload:object | null) => void
+}
+export const createAuthSlice: StateCreator<AuthSlice> = (set) => ({
```

Comment:
n Zustand, there is no concept of 'slice'. When you create a store, it returns a hook. I noticed that you wrote actions differently than in Redux, but the main reason to use Zustand is to make every store independent.

Reference 
https://dev.to/franklin030601/using-zustand-with-react-js-9di

Reply:
Okay. But Zustand kept an example of slice pattern in their official doc-  https://docs.pmnd.rs/zustand/guides/typescript#slices-pattern . Anyway it's also great to use independent store.